org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/configurator/MojoExecutionKey.java

```diff
@@ -9,113 +9,25 @@
-  private static final long serialVersionUID = -2074582830199438890L;
-
-  private final String groupId;
-
-  private final String artifactId;
-
-  private final String version;
-
-  private final String goal;
-
-  private final String executionId;
-
-  private final String lifecyclePhase;
+public record MojoExecutionKey(String groupId, String artifactId, String version, String goal, String lifecyclePhase,
+    String executionId) implements Serializable {
 
   public MojoExecutionKey(MojoExecution mojoExecution) {
-    groupId = mojoExecution.getGroupId();
-    artifactId = mojoExecution.getArtifactId();
-    version = mojoExecution.getVersion();
-    goal = mojoExecution.getGoal();
-    executionId = mojoExecution.getExecutionId();
-    lifecyclePhase = mojoExecution.getLifecyclePhase();
-  }
-
-  public MojoExecutionKey(String groupId, String artifactId, String version, String goal, String lifecyclePhase,
-      String executionId) {
-    this.groupId = groupId;
-    this.artifactId = artifactId;
-    this.version = version;
-    this.goal = goal;
-    this.lifecyclePhase = lifecyclePhase;
-    this.executionId = executionId;
-  }
-
-  @Override
-  public int hashCode() {
-    int hash = groupId.hashCode();
-    hash = 17 * hash + artifactId.hashCode();
-    hash = 17 * hash + version.hashCode();
-    hash = 17 * hash + goal.hashCode();
-    hash = 17 * hash + (executionId != null ? executionId.hashCode() : 0);
-    hash = 17 * hash + (lifecyclePhase != null ? lifecyclePhase.hashCode() : 0);
-    return hash;
-  }
-
-  @Override
-  public boolean equals(Object obj) {
-    if(this == obj) {
-      return true;
-    }
-    return obj instanceof MojoExecutionKey other && //
-        Objects.equals(groupId, other.groupId) && Objects.equals(artifactId, other.artifactId)
-        && Objects.equals(version, other.version) && Objects.equals(goal, other.goal)
-        && Objects.equals(executionId, other.executionId) && Objects.equals(lifecyclePhase, other.lifecyclePhase);
-  }
-
-  public String getGroupId() {
-    return this.groupId;
-  }
-
-  public String getArtifactId() {
-    return artifactId;
-  }
-
-  public String getVersion() {
-    return version;
-  }
-
-  public String getGoal() {
-    return goal;
-  }
-
-  public String getExecutionId() {
-    return executionId;
-  }
-
-  public String getLifecyclePhase() {
-    return lifecyclePhase;
+    this(mojoExecution.getGroupId(), mojoExecution.getArtifactId(), mojoExecution.getVersion(), mojoExecution.getGoal(),
+        mojoExecution.getLifecyclePhase(), mojoExecution.getExecutionId());
   }
 
   public String getKeyString() {
     return groupId + ":" + artifactId + ":" + version + ":" + goal + ":" + executionId + ":" + lifecyclePhase;
   }
-
-  @Override
-  public String toString() {
-    return groupId + ":" + artifactId + ":" + version + ":" + goal + " (execution: " + executionId + ", phase: "
-        + lifecyclePhase + ")";
-  }
-
-  public boolean match(MojoExecution mojoExecution) {
```

Comment:
Please note that `match` has slightly different semantics than `equals` so unless the record is not `final` we should not just remove this method!

Reply:
> Please note that `match` has slightly different semantics than `equals` so unless the record is not `final` we should not just remove this method! (or is this method not used anywhere in the code?)

Records are always final AFAIK. The only difference to equals is that `match()` ignores the `lifecyclePhase`.

It is only used in `MavenProjectFacade` and I therefore moved it there.

It's a bit hard to tell, if such helper methods should remain in the API or not. It depends on if there are real use-cases for downstream consumers, but I don't know that.